client/lib/feed-post-store/index.js

```diff
@@ -198,7 +198,7 @@ function receivePostFromPage( newPost ) {
 }
 
 function receivePost( feedId, postId, post ) {
-	if ( post.errors ) {
+	if ( post.errors || post.status_code === 404 ) {
```

Comment:
@TooTallNate:

> The batch API endpoint now gives an object of { status_code: 404 } for non-existing API endpoints, as expected.

That's nice. However, it doesn't give the `errors` object, which is expected. To countermeasure, let's look for just `status_code` as well.